onnxruntime/core/providers/cpu/generator/range.cc

```diff
@@ -80,24 +81,43 @@ static Status ComputeRange(OpKernelContext* ctx) {
   return Status::OK();
 }
 
+namespace range_internal {
+
+// Workaround GCC bug https://gcc.gnu.org/bugzilla/show_bug.cgi?id=47226
+// can not specify lambda directly
+template <class T, class... Args>
+inline int invoke_computerange_callable(size_t& called, Status& s, int32_t dt_type, Args&&... args) {
+  auto fn = [&] {
+    if (utils::ToTensorDataType<T>() == dt_type) {
+      s = ComputeRange<T>(std::forward<Args>(args)...);
+      ++called;
+    }
+    return 0;
+  };
+  return fn();
+}
+
+
+template<typename... Types>
+inline Status CallDispatcher(int32_t dt_type, OpKernelContext* ctx) {
+  Status s;
+  size_t called = 0;
+
+  int results[] = {0, invoke_computerange_callable<Types>(called, s, dt_type, ctx)...};
+
```

Comment:
Is there a benefit to returning a value and populating results[] vs. returning void from invoke_computerange_callable?

Reply:
Parameter pack expansion is permitted only in certain contexts. Folding expression is only supported in C++17 and with our compiler we will not get that ever. Interger array context seemed to be the least confusing and performant.

---
In reply to: [341942513](https://github.com/microsoft/onnxruntime/pull/2307#discussion_r341942513) [](ancestors = 341942513)